lib/github/commands/man2html

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+what=${1:--}
+cat "$what" | mandoc -Kutf-8 -Thtml | xmlstarlet tr $(dirname "$0")/man2html.xslt | sed 's/<\/\?html>//g' | sed 's/<\/\?body>//g'
```

Comment:
Wouldn't it make more sense to use `mandoc -Tmarkdown` for output?

Reply:
@Alhadis We thought about that, but according to http://mandoc.bsd.lv/man/mandoc.1.html:

> Markdown is a very weak markup language, so all semantic markup is lost, and even part of the presentational markup may be lost. Do not use this as an intermediate step in converting to HTML; instead, use -T html directly.

Reply:
Yes, but that's written with the assumption that HTML is the targeted output format. If the destination format is Markdown, that's another matter entirely.

Reply:
The destination format is not Markdown, though? `github/markup` emits HTML, there doesn't need to be an intermediate Markdown step.

Reply:
Oh! That changes things. Sorry, I was somehow under the impression that rendered documents were coerced to markdown before processing. My bad.

In that case, 👌

Reply:
Nope, we take HTML directly and then run it through https://github.com/jch/html-pipeline

Reply:
How about

    cat "$@" |
    mandoc -Kutf-8 -Thtml |
    xmlstarlet tr "$(dirname "$0")/man2html.xslt" |
    sed 's,</*html>,,g; s,</*body>,,g'

`$0`'s dirname might contain a space. POSIX sed's Basic Regular Expressions don't support question mark.

Reply:
Hi,
mandoc upstream maintainer speaking here.  @Alhadis just told me about this project, and i'm intrigued to see it.  If you have any particular concerns about the HTML code emitted from mandoc, tell me and i shall see what i can do about them - though obviously i cannot promise deadlines for fixing stuff.

Do i understand correctly that you want HTML output without the enclosing html and body tags?
That is exactly what the mandoc output option "-O fragment" does.
If you say "mandoc -T html -O fragment ls", you get something that starts with "table class=head" and ends with the closing table tag of the foot table.
Of course, you also lose the DOCTYPE and the head that way.

Second point, "-K utf-8" is almost certainly a bad idea.  If the input file is indeed valid UTF-8, mandoc will treat it correctly without that option because UTF-8 is mandoc's preferred input encoding.  If the input file is something else, it is most likely ISO-LATIN-1 from what i have seen in practice; in that case, the option breaks the encoding whereas mandoc would treat the input correctly without the option.  If the input is a different legacy encoding, mandoc cannot handle it either way, the file would have to be converted with iconv(1) first.  The only case where the option might theoretically make sense is when the author *intended* to write UTF-8, but somehow managed to screw up the encoding of the very first non-ASCII character in the file to not be a valid UTF-8 encoding.  But that is really more a theoretical than a practical consideration and certainly not worth breaking ISO-LATIN-1 input support, which still occurs quite a bit in real-world manual pages.

Third point, even less important, what is the point of using cat(1) here?
Why not just "mandoc -Kutf-8 -Thtml $what | xmlstarlet ..."?
If it is one file, that is simpler and does the same.
If it is more than one file, mandoc *must* be called that way.  Something like "cat man.1 apropos.1 | mandoc ..." works rather poorly because the parser state cannot be reset between the two manual pages, whereas "mandoc ... man.1 apropos.1" does correct parsing.